### ISSUE TITLE
Unexpected results from element selection after walking into iframe

### ISSUE THREAD

**Issue by katayama-lepidum** [watir/watir-webdriver#361]
I'm using watir-webdriver 0.8.0 on Ubuntu 14.04.3, Ruby 2.2.1 and headless 2.2.0.

After enumerating links inside iframe, Watir::Browser#links and #iframes returns unexpected value.
Also, call of Watir::Browser#inspect makes them back to work properly.

Is this behavior is expected one or not?

Test page is here: http://katayama-lepidum.github.io/watir-iframe/

```
$ bundle exec irb -rwatir-webdriver -rheadless
2.2.1 :001 > h = Headless.new
 => #<Headless:0x9ef5db8 @display=99, @xvfb_launch_timeout=10, @autopick_display=true, @reuse_display=true, @dimensions="1280x1024x24", @video_capture_options={}, @destroy_at_exit=true>
2.2.1 :002 > h.start
 => #<Proc:0x9ef3748@/home/katayama/work/a/vendor/bundle/ruby/2.2.0/gems/headless-2.2.0/lib/headless.rb:224>
2.2.1 :003 > b = Watir::Browser.new :chrome
 => #<Watir::Browser:0x..f38659b4 url="data:," title="data:,">
2.2.1 :004 > b.goto 'http://katayama-lepidum.github.io/watir-iframe/'
 => "http://katayama-lepidum.github.io/watir-iframe/"
2.2.1 :005 > b.links.to_a
 => [#<Watir::Anchor:0x..fdbc3a2ec located=false selector={element: (webdriver element)}>, #<Watir::Anchor:0x5350f058 located=false selector={element: (webdriver element)}>]
2.2.1 :006 > b.iframes.to_a
 => [#<Watir::IFrame:0x..fafe5e1a0 located=false selector={:tag_name=>"iframe", :index=>0}>]
2.2.1 :007 > b.iframes[0].links.each do |e| p e end
 => []
2.2.1 :008 > b.links.to_a        # <===== expecting 2 elements, but empty array
 => []
2.2.1 :009 > b.iframes.to_a        # <===== expecting single element, but empty array
 => []
2.2.1 :010 > b.inspect
 => "#<Watir::Browser:0x..f38659b4 url=\"http://katayama-lepidum.github.io/watir-iframe/\" title=\"Watir IFRAME\">"
2.2.1 :011 > b.links.to_a
 => [#<Watir::Anchor:0x5a2e0e58 located=false selector={element: (webdriver element)}>, #<Watir::Anchor:0x..f8557fc78 located=false selector={element: (webdriver element)}>]
2.2.1 :012 > b.iframes.to_a
 => [#<Watir::IFrame:0x7f802fac located=false selector={:tag_name=>"iframe", :index=>0}>]
2.2.1 :013 >
```

**Comment by p0deje** [comment]
@katayama-lepidum Thanks for the report!

@titusfortner I believe you know better the way iframe locator is implemented. Care to take a look into this?

**Comment by titusfortner** [comment]
Oh interesting. I bet element collections aren't switching browser contexts like we have single element location doing. I'll take a look tomorrow.

**Comment by titusfortner** [comment]
Fixed with - 75fb34a

It adds a little bit of overhead in calling nested element collections because it is going to verify the presence of all of the parents up to the browser, but you can now reference just element collections from any browser context in any order.

**Comment by katayama-lepidum** [comment]
Works fine, Thank you!
